src/fulltrade/FullTradeCard.js

```diff
@@ -90,12 +89,10 @@ export default class FullTradeCard extends Component {
 
         return (
             <div disabled={disabled} className={'trade-panel'}>
-                <Modal
-                    shown={!!purchaseInfo.buy_error}
-                    onClose={() => actions.updatePurchaseInfo(index, 'buy_error', undefined)}
-                >
-                    <PurchaseFailed failure={purchaseInfo.buy_error} />
-                </Modal>
+
+                {!!purchaseInfo.buy_error &&
+                    showError(purchaseInfo.buy_error.code + ' ' + purchaseInfo.buy_error.message)
```

Comment:
Oh i forgot showError is not a component . Nonetheless @borisyankov wouldn't The current Modal be more appropriate for our need for this case? Have analyse through all the Modals and thought this qualifies for the showError util , considering its being render in the render method wouldn't using the modal be more appropriate , less we are going to take it to another method. Have thought a lot but the requirement not well clear.